pkg/cloud/cloud.go

```diff
@@ -1,37 +1,429 @@
+	for _, configOption := range fileCacheOptions.LustreConfiguration {
+		configSplit := strings.SplitN(configOption, "=", 2)
+		configKey := configSplit[0]
+		configValue := configSplit[1]
+		configMap[configKey] = configValue
+	}
+
+	if deploymentType, ok := configMap[LustreConfigOptionsDeploymentType]; ok {
+		lustreConfiguration.SetDeploymentType(deploymentType)
+	} else {
+		lustreConfiguration.SetDeploymentType(DefaultFileCacheDeploymentType)
+	}
+
+	if metadataConfiguration, ok := configMap[LustreConfigOptionsMetaDataConfiguration]; ok {
+		storageCapacityPair := strings.Split(metadataConfiguration[1:len(metadataConfiguration)-1], "=")
+		metadataStorageCapacity, err := strconv.ParseInt(storageCapacityPair[1], 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid input for MetadataConfiguration StorageCapacity: %s", err)
+		}
+		metadataConfig := &fsx.FileCacheLustreMetadataConfiguration{StorageCapacity: aws.Int64(metadataStorageCapacity)}
+		lustreConfiguration.SetMetadataConfiguration(metadataConfig)
+	} else {
+		metadataConfig := &fsx.FileCacheLustreMetadataConfiguration{StorageCapacity: aws.Int64(DefaultMetadataStorageCapacity)}
+		lustreConfiguration.SetMetadataConfiguration(metadataConfig)
+	}
+
+	if perUnitStorageThroughput, ok := configMap[LustreConfigOptionsPerUnitStorageThroughput]; ok {
+		throughput, err := strconv.ParseInt(perUnitStorageThroughput, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid input for MetadataConfiguration PerUnitStorageThroughput: %s", err)
+		}
+		lustreConfiguration.SetPerUnitStorageThroughput(throughput)
+	} else {
+		lustreConfiguration.SetPerUnitStorageThroughput(DefaultPerUnitStorageThroughput)
+	}
+
+	if fileCacheOptions.WeeklyMaintenanceStartTime != "" {
+		lustreConfiguration.SetWeeklyMaintenanceStartTime(fileCacheOptions.WeeklyMaintenanceStartTime)
+	}
+
+	input.SetLustreConfiguration(lustreConfiguration)
 
-	metadata, err := NewMetadataService(svc)
+	if fileCacheOptions.KmsKeyId != "" {
+		input.SetKmsKeyId(fileCacheOptions.KmsKeyId)
+	}
+
+	if fileCacheOptions.CopyTagsToDataRepositoryAssociations {
+		input.SetCopyTagsToDataRepositoryAssociations(true)
+	}
+
+	klog.V(4).Infof("CreateFileCacheInput: ", input.GoString())
+	output, err := c.fsx.CreateFileCacheWithContext(ctx, input)
 	if err != nil {
-		return nil, fmt.Errorf("could not get metadata from AWS: %v", err)
+		if isIncompatibleParameter(err) {
+			return nil, ErrFcExistsDiffSize
+		}
+		return nil, fmt.Errorf("CreateFileCache failed: %v", err)
 	}
 
-	return &cloud{
-		metadata: metadata,
+	if output.FileCache.LustreConfiguration.MountName == nil {
+		return nil, fmt.Errorf("mountName returned in CreateFileCache is nil")
+	}
+	mountName := *output.FileCache.LustreConfiguration.MountName
+
+	perUnitStorageThroughput := int64(0)
+	if output.FileCache.LustreConfiguration.PerUnitStorageThroughput != nil {
+		perUnitStorageThroughput = *output.FileCache.LustreConfiguration.PerUnitStorageThroughput
+	}
```

Comment:
If this is nil, would we want to throw an error or warning here? Would be weird to see perUnitStorageThroughput of 0 without any additional messages